include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -0,0 +1,176 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_parallel_backend_sycl_radix_sort_one_wg_H
+#define _ONEDPL_parallel_backend_sycl_radix_sort_one_wg_H
+
+//namespace oneapi
+//{
+//namespace dpl
+//{
+//namespace __par_backend_hetero
+//{
+
+template <int ITEMS_PER_THREAD, typename KeyT, typename _Wi, typename _Src, typename _Keys>
+void
+__block_load(const _Wi __wi, const _Src& __src, _Keys& __keys, const uint32_t __n)
+{
+    constexpr KeyT __default_key = (KeyT)0xffffffff;
```

Comment:
This might differ depending on the sorting order and the type.
For example, I would expect 0 for descending order and unsigned type.

Reply:
I think it doesn't matter, because we ignore values out of a data iteration space.
BTW, Currently, we have `constexpr KeyT __default_key = KeyT{};`

Reply:
I guess it does matter. Data iteration space is verified twice:
-  when the data is put into `keys` array (before all radix sort iterations)
-  when the data is put back to `__src` (after all radix sort iterations).

Each radix sort iteration shuffles elements when writing then into a temporary array in `__to_blocked` function. With the use of `0xffffffff` value with ascending soring and unsigned keys, the values were always put at the end of the temporary array and then were not written to `__src` because of `__r < __n` check.

To always put the `default_key` values at the end of the temporary array, its value should depend on:
 - if the keys are signed or unsigned
 - sorting order
 - probably if the keys have integer or floating point type (depends on their encoding).

Reply:
Fixed.